pkg/controllers/management/node/controller.go

```diff
@@ -499,6 +503,11 @@ func (m *Lifecycle) saveConfig(config *nodeconfig.NodeConfig, nodeDir string, ob
 		return obj, err
 	}
 
+	pool, err := m.getNodePool(obj.Spec.NodePoolName)
```

Comment:
What happens when the nodePoolName is empty?

Reply:
It won't. The `saveConfig` function will be only run for the node created by node pool. It's controlled by condition `NodeConditionProvisioned` and this condition will be set to true in custom node and cloud k8s node.